aci/resource_aci_fvnsmcastaddrinstp.go

```diff
@@ -0,0 +1,307 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/aci-go-client/client"
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAciMulticastAddressPool() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAciMulticastAddressPoolCreate,
+		UpdateContext: resourceAciMulticastAddressPoolUpdate,
+		ReadContext:   resourceAciMulticastAddressPoolRead,
+		DeleteContext: resourceAciMulticastAddressPoolDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAciMulticastAddressPoolImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: AppendAttrSchemas(map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"multicast_address_block": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: AppendAttrSchemas(map[string]*schema.Schema{
+						"dn": {
```

Comment:
why do we need the dn?

Reply:
I use the dn to do a delete of the object. On changes I need to delete address block object first to assure change succeeds. An error can occur when you want to update the block range with an overlapping range, thus you cannot update directly but you will need to remove the object and apply new object in two terraform apply actions.

This could also be done with looking at the from and to inputs but think that would make things more complicated.

Reply:
add comment this is internally used